src/Features/VisualBasic/Portable/CodeFixes/PreferFrameworkType/VisualBasicPreferFrameworkTypeCodeFixProvider.vb

```diff
@@ -0,0 +1,15 @@
+﻿'Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Composition
+
+Namespace Microsoft.CodeAnalysis.CodeFixes.PreferFrameworkType
+
+    <ExportCodeFixProvider(LanguageNames.VisualBasic, Name:=PredefinedCodeFixProviderNames.PreferFrameworkType), [Shared]>
+    Friend Class VisualBasicPreferFrameworkTypeCodeFixProvider
+        Inherits AbstractPreferFrameworkTypeCodeFixProvider
+
+        Protected Overrides Function GenerateTypeSyntax(symbol As ITypeSymbol) As SyntaxNode
+            Return symbol.GenerateTypeSyntax()
```

Comment:
Oh.  You don't need ISemanticFacts for this.  Just get  SyntaxGenerator for your document and ask to create the syntax for the type symbol.  We already have a public API for this.